BritneyS/GetFed
# Custom entry list

## What you did :question:
- Custom user entries added to tableview (initially and during search)
- Added icon on food listings to indicate where entry was added by user
- Added image credit to README

## How you did it :white_check_mark:
- Made separate array for custom food entries and data filtered in search
- `foodEntries` array populated in initial retrieval of custom entry Core Data records
- `filteredFoodArray` is populated based on term in search bar
- `foodArray` is populated with `filteredFoodArray` and API call results if API call is made (upon tapping search button/pushing `return`)
- Added `user-created.png` icon and `blank-tranparent.png` icon to toggle between (on tableview), based on whether the food listing is a custom food entry or a API call result
- Added `isUserAdded` property in `Food` that is changed to `true` when `foodEntries` array is populated
- Wrote method `showImage` in FoodResultTableViewCell.swift that is called in `cellForRowAt` that shows `user-created.png` icon in the cell if `isUserAdded` is `true` and `blank-tranparent.png` is false
- Added image string literals to identity enum
- Included image credit in README with link to author page and image

## How to test it :microscope:
- Run the app
- Tap 'Food Search'
- Note that all visible custom food entries have an icon displayed next to them in a cell
- Search for 'Potato salad'
- Note that only the relevant search result from the Core Data (custom user entry) records are returned
- Press `return`
- Note that the top result from Core Data has an icon displayed in its cell, and the results from the API call do not


## Screenshots (if applicable) :camera:
![simulator screen shot - iphone xr - 2018-12-18 at 14 35 23](https://user-images.githubusercontent.com/8409475/50179669-9b66b580-02d5-11e9-9be6-627672f4ff1a.png)
![simulator screen shot - iphone xr - 2018-12-18 at 14 35 30](https://user-images.githubusercontent.com/8409475/50179678-a0c40000-02d5-11e9-915b-3898f7f8cab9.png)
![simulator screen shot - iphone xr - 2018-12-18 at 14 35 52](https://user-images.githubusercontent.com/8409475/50179681-a4f01d80-02d5-11e9-8637-bf77424b9b23.png)
